client/dashboard/app/notifications/index.tsx

```diff
@@ -2,15 +2,25 @@ import { useNavigate } from '@tanstack/react-router';
 import { Button, Dropdown } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { bellUnread, bell } from '@wordpress/icons';
-import { Suspense, lazy } from 'react';
+import clsx from 'clsx';
+import { Suspense, lazy, useState } from 'react';
 import wpcom from 'calypso/lib/wp';
+import { useAuth } from '../auth';
+import './style.scss';
 
 const AsyncNotificationApp = lazy( () => import( '@automattic/notifications/src/app' ) );
 
 export default function Notifications( { className }: { className: string } ) {
 	const navigate = useNavigate();
+	const { user } = useAuth();
+	const [ hasUnseenNotifications, setHasUnseenNotifications ] = useState( user.has_unseen_notes );
 
 	const actionHandlers = ( onClosePanel: () => void ) => ( {
+		APP_RENDER_NOTES: [
+			( store: any, { newNoteCount }: { newNoteCount: number } ) => {
+				setHasUnseenNotifications( newNoteCount > 0 );
```

Comment:
I guess we should also set the value to false when the user opens the notification 🙂